app/controllers/rooms_controller.rb

```diff
@@ -24,17 +22,37 @@ def create
 
   def destroy
     room = Room.find(room_params[:id])
-    render json: room
-    room.users.destroy
-    room.user_rooms.destroy
-    room.room_questions.destroy
     room.destroy
+
+    render json: room
```

Comment:
You almost always want to render last in a function. If you do it before, and there's an error after your render call, you could end up trying to render an error to the client, causing a double render error.